libs/payments/eligibility/src/lib/eligibility.manager.ts

```diff
@@ -54,4 +67,145 @@ export class EligibilityManager {
     }
     return result;
   }
+
+  async getPlanByInterval(offeringConfigId: string, interval: string) {
+    const targetOffering = (
+      await this.contentfulManager.getEligibilityContentByOffering(
+        offeringConfigId
+      )
+    ).getOffering();
+
+    // make a helper in util?
+    const targetPlanIds = targetOffering.defaultPurchase.stripePlanChoices;
+
+    const plans = [];
+    for (const planId of targetPlanIds) {
+      const plan = await this.stripeManager.getPlan(planId);
+      if (plan.interval === interval) {
+        plans.push(plan);
+      }
+    }
+
+    // ?? - could multiple plans have the same interval;
+    // interval values - 'daily', 'weekly', 'monthly', '6monthly', 'yearly'
+    return plans[0];
+  }
+
+  /**
+   * Checks if user is eligible to subscribe to plan
+   */
+  async checkEligibility(
+    interval: string,
+    offeringConfigId: string,
+    stripeCustomerId?: string | null
+  ) {
+    if (!stripeCustomerId) {
+      return {
+        state: CartState.START,
+        errorReasonId: null,
+      };
+    }
+
+    const targetPlan = await this.getPlanByInterval(offeringConfigId, interval);
```

Comment:
I don't think we should be fetching the `targetPlan` at this point.

`getOfferingOverlap` currently only uses `targetPlan` to fetch the correct `targetOffering` from the CMS, so why not just pass in the `targetOffering` in directly?

Then `targetPlan` can be fetched only when it's necessary, i.e. when comparison === SAME.